src/OSPSuite.Core/Domain/Services/EventBuilderTask.cs

```diff
@@ -94,20 +94,24 @@ private void createEventGroupFrom(EventGroupBuilder eventGroupBuilder, ModelConf
       /// </summary>
       private void createEventGroupInContainer(EventGroupBuilder eventGroupBuilder, IContainer sourceContainer, ModelConfiguration modelConfiguration, EventGroupBuildingBlock eventGroupBuildingBlock, MergeBehavior mergeBehavior)
       {
+         modelConfiguration.UpdateReplacementContext();
+         var (_, simulationBuilder, _) = modelConfiguration;
+
          //this creates recursively all event groups for the given builder
-         var (_, simulationBuilder, replacementContext) = modelConfiguration;
          var eventGroup = _eventGroupMapper.MapFrom(eventGroupBuilder, simulationBuilder);
 
          tryMergeEventGroupInContainer(sourceContainer, eventGroup, mergeBehavior, eventGroupBuildingBlock);
 
-         //needs to add the required transport into model only for the added event group
+         // after the Applications container is added to the model, update the replacement context
+         modelConfiguration.UpdateReplacementContext();
```

Comment:
This is the important missing line of code

Reply:
Why do we do it twice? line 97 and line 106?

Reply:
ah, I don't need it in line 97